### PR TITLE
[Macros] Don't allow types to shadow freestanding macros.

### PR DESCRIPTION
**Pull request by DougGregor**
When performing name lookup for freestanding macros (e.g., after the `#`), don't allow types to shadow macros from imported libraries.

Fixes rdar://110429368.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test

File: test/Macros/macro_shadowing.swift [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+// 
+// RUN: %target-typecheck-verify-swift -swift-version 5 -load-plugin-library %t/%target-library-name(MacroDefinition) -module-name MacroUser -I %t
+
+import freestanding_macro_library
+
+struct stringify<T> { }
+
+func testStringify(a: Int, b: Int) {
+  _ = #stringify(a + b)
```
Comment (bnbarham) [inline test/Macros/macro_shadowing.swift:19]: What's the diagnostic now if there's a type but no macro? I assume just a "cannot find macro..."?

Comment (DougGregor) [reply test/Macros/macro_shadowing.swift:19]: Yes, the same as it was before, because everything that isn't a macro gets filtered out.

**Review by hborla — APPROVED** [review]

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Pull request merged**